src/cappa/output.py

```diff
@@ -103,6 +142,22 @@ def __init__(
         message: list[Displayable] | Displayable,
         *,
         code: str | int | None = 0,
+        prog: str | None = None,
     ):
-        super().__init__(code=code)
+        super().__init__(code=code, prog=prog)
         self.message = message
+
+
+def rich_to_ansi(
+    console: Console, message: list[Displayable] | Displayable | str
+) -> str:
+    with console.capture() as capture:
+        if isinstance(message, list):
+            *messages, last = message
+            for m in messages:
+                console.print(m)
+            console.print(last, end="")
+        else:
+            console.print(message, end="")
```

Comment:
IIUC, the logic here is to print each element except the last with a new line at the end, and the last one without a newline at the end? Since you strip the result, couldn't you just `console.print(message)` each, without worrying about `end=""`?

Reply:
mmm good point. i added the `strip` call after writing all this, because certain elements like tables include newlines automatically, and i couldn't get a 1 newline at the end reliably. I can almost certainly remove this bit